### DISCUSSION TITLE
How to mask dynamic url in Next13( with app router)

### DISCUSSION THREAD

**Discussion by SaXaPhonist** [vercel/next.js#51338] [Help]
### Summary

I need to mask dynamic url for example part with /about/[id] -> to /about/some-readable-text

**Comment by devjiwonchoi** [comment]
The `App Router` suggests Dynamic Route be declared as a folder name, which differs from the `Page Router` being filename.

> A Dynamic Segment can be created by wrapping a folder's name in square brackets: [folderName]

Note that the `page file` must be named `page.js | jsx | tsx`.

> A page is UI that is unique to a route. You can define pages by exporting a component from a page.js file.

## Conclusion

In your case, the project route can be something like `app/about/[id]/page.tsx`.

## Reference
- [Dynamic Routes Convention](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes#convention)
- [Pages](https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#pages)
- [page.js](https://nextjs.org/docs/app/api-reference/file-conventions/page)

**Reply by SaXaPhonist** [reply]
@devjiwonchoi  i think it's a little different from what i what. In my case i already have folders route structure like:

    `/about
      page.tsx. 
      /[slug].  
          page.tsx` 
i want to approach behavior when you click link in about page to [slug] page, slug remains  === id (e.x `/about/7`) for making request for data and at the same time browser shows in url some readable title ( e.x.  `/about/company`)

**Reply by devjiwonchoi** [reply]
Could it be just `/about/company/1` or just `/about/company`?

**Reply by SaXaPhonist** [reply]
i have to use only "id" to get data from backend, so i can't just replace to "title" instead
but the view of url for user should be readable "title"

**Reply by SaXaPhonist** [reply]
@devjiwonchoi  or in other words i have to pass id through navigation to get data, but it will be not good for me if i just pass id as searchParam cause it show  url view as '/about/title?id=2', my goal is '/about/title'

**Reply by devjiwonchoi** [reply]
In your case, you can use the `as` feature in the `Link` Component.

```js
import Link from 'next/link'
export default async function Home() {
  // get id from API
  const id = 1
  return (
    <>
      <h1>This is Home</h1>
      <Link
        href={{
          pathname: '/about/company/[id]',
          query: { id: id },
        }}
        as='/about/company'
      >
        Go to About Company Dynamic Route Page as /about/company
      </Link>
    </>
  )
}
```

It directs you to `/about/company/[id]` as the URL displayed as `/about/company`.

## Reference
- [Example of `as` with Middleware](https://nextjs.org/docs/pages/api-reference/components/link#middleware)
